rosbag2_cpp/test/rosbag2_cpp/test_compression_options.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include <string>
+
+#include "rosbag2_cpp/compression_options.hpp"
+
+TEST(CompressionOptionsFromStringTest, BadInputReturnsNoneMode)
+{
+  std::string compression_mode_string{"bad_mode"};
+  auto compression_mode = rosbag2_cpp::compression_mode_from_string(compression_mode_string);
+  EXPECT_EQ(compression_mode, rosbag2_cpp::CompressionMode::NONE);
+}
+
+TEST(CompressionOptionsFromStringTest, FileStringReturnsFileMode)
+{
+  std::string compression_mode_string{"file"};
+  auto compression_mode = rosbag2_cpp::compression_mode_from_string(compression_mode_string);
+  EXPECT_EQ(compression_mode, rosbag2_cpp::CompressionMode::FILE);
+}
+
+TEST(CompressionOptionsFromStringTest, MessageStringReturnsMessageMode)
+{
+  std::string compression_mode_string{"MESSAGE"};
+  auto compression_mode = rosbag2_cpp::compression_mode_from_string(compression_mode_string);
+  EXPECT_EQ(compression_mode, rosbag2_cpp::CompressionMode::MESSAGE);
+}
+
```

Comment:
Add a unit test for `compression_mode_to_string` passed an invalid `CompressionMode` 

I think you can do:
```c++
const auto compression_mode = static_cast<CompressionMode>(100);
```